book/layout.md

```diff
@@ -201,113 +213,72 @@ BLOCK_ELEMENTS = [
 ]
 ```
 
-We'll use use `BlockLayout` for elements whose children are in that
-list, and `InlineLayout` otherwise. Let's put that logic in a new
-`layout_mode` function:
+We'll use `BlockLayout` for elements with children from that list, and
+`InlineLayout` otherwise. Put that logic in a new `layout_mode`
+function:
 
 ``` {.python}
 def layout_mode(node):
-    has_text = False
-    has_containers = False
-    for child in node.children:
-        if isinstance(child, Text):
-            has_text = True
-        elif child.tag in BLOCK_ELEMENTS:
-            has_containers = True
-        else:
-            has_text = True
-    if has_containers or not has_text:
-        return "block"
-    else:
+    if isinstance(node, Text):
         return "inline"
+    elif node.children:
+        for child in node.children:
+            if child.tag in BLOCK_ELEMENTS:
+                return "block"
+        return "inline"
+    else:
+        return "block"
 ```
 
-`BlockLayout` can now create child layout objects by looping over its
-children, calling `layout_mode` to determine what type of layout
-object to create:
+The cases make sure text nodes get inline layout while empty elements
+get block layout. Now we can call `layout_mode` to determine which
+layout mode to use for each element:
 
 ``` {.python}
 class BlockLayout:
     def layout(self):
         previous = None
         for child in self.node.children:
             if layout_mode(child) == "inline":
-                previous = InlineLayout(child, self, previous)
+                next = InlineLayout(child, self, previous)
```

Comment:
Is this fixing a bug in the old code?

Reply:
No—it's just clarifying. Before you create a new layout object by referring to the previous node, then store the new node in the `previous` variable. Now, you store it in a new `next` variable which is later saved into `previous`.